### PR TITLE
QT4 EL8 Changes

### PR DESCRIPTION
**Pull request by tst-cjeffries**
Same as previous PR, but with squashed commits.

### CODE REVIEW

File: earth_enterprise/src/SConstruct [hunk 1]
```diff
@@ -277,12 +280,29 @@ else:
...
+                '-Wno-error=class-memaccess',
+                '-Wno-error=memset-elt-size',
+                '-Wno-error=address',
+                '-Wno-error=unused-local-typedefs']
+
+# Determine libpng version based on el version
+if el_ver == "el8":
+  cppflags += ['-DUSING_LIBPNG16']
+else:
+  cppflags += ['-DUSING_LIBPNG15']
```
Comment (dyfet) [inline earth_enterprise/src/SConstruct:304]: So this would include Ubuntu....

File: earth_enterprise/src/common/gedbroot/SConscript [hunk 2]
```diff
@@ -38,6 +38,11 @@ gedbroot = env.sharedLib('gedbroot',
                                'QtCore'
                                ])
 
+if env['el_ver'] == 'el8':
+  png_ver = 'png16'
+else:
+  png_ver = 'png15'
```
Comment (tst-dsugar) [inline earth_enterprise/src/common/gedbroot/SConscript:44]: So this would cover ubuntu 18.04/16.04 too...?

Comment (tst-cjeffries) [reply earth_enterprise/src/common/gedbroot/SConscript:44]: Assuming that it uses png15. We may need a slight adjustment in logic if we go with 16. Something along the lines of `if not env['el_ver'] or env['el_ver'] == 'el8':`.

Comment (tst-basilhuffman) [reply earth_enterprise/src/common/gedbroot/SConscript:44]: i think it would be cleaner to make `png_ver` an environment variable instead of `el_ver`

**Comment by tst-basilhuffman** [comment]
looks good, just had that one comment

File: earth_enterprise/src/common/gedbroot/SConscript (hunk 2 above)
Comment (tst-cjeffries) [reply earth_enterprise/src/common/gedbroot/SConscript:44]: I agree with that. I thought of that after I had finished up.

**Pull request merged**